src/java/htsjdk/samtools/cram/encoding/huffman/HuffmanParamsCalculator.java

```diff
@@ -0,0 +1,135 @@
+package htsjdk.samtools.cram.encoding.huffman;
+
+import htsjdk.samtools.cram.common.MutableInt;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * A utility class to calculate Huffman encoding parameters based on the values to be encoded.
+ */
+public class HuffmanParamsCalculator {
```

Comment:
@cmnbroad not sure if the class is needed at all. This is intended for advanced users who choose to override default compression but it seems marginally useful. I'll remove the class and the tests.